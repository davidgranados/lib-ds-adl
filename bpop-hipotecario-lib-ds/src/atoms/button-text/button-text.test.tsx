import React from "react";
import { mount, ReactWrapper } from "enzyme";

import ButtonText from "./button-text";

describe("<ButtonText />", () => {
  let wrapper: ReactWrapper<typeof ButtonText>;
  const variant = "primary";
  const text = "button-text";
  const onFocus = jest.fn();
  const onBlur = jest.fn();
  const onClick = jest.fn();
  beforeEach(() => {
    jest.clearAllMocks();
    // given
    wrapper = mount<typeof ButtonText>(
      <ButtonText text={text} variant={variant} onFocus={onFocus} onBlur={onBlur} onClick={onClick} />
    );
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should handle focus", () => {
    // when
    wrapper.find("button").simulate("focus");
    // then
    expect(onFocus).toBeCalled();
  });

  it("should handle blur", () => {
    // when
    wrapper.find("button").simulate("blur");
    // then
    expect(onBlur).toBeCalled();
  });

  it("should call onClick", () => {
    // when
    wrapper.find("button").simulate("click");
    // then
    expect(onClick).toBeCalled();
  });
});
