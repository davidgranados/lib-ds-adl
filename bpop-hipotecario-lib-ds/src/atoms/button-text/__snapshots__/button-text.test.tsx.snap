// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonText /> should match snapshot 1`] = `
<button
  className="default font-size-md default primary"
  disabled={false}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onFocus={[Function]}
  type="button"
>
  button-text
</button>
`;
